fix: make cli skip private packages without version field

bhovhannes opened the pull request:
This closes #620 

Per conversation there, it was decided to skip packages without "version" field in CLI. I limited that to _private_ packages without "version" field to limit the blast radius of this behavior change.

bhovhannes commented:
@Andarist I'll be grateful for you review. I believe this does not contradict with #705 .

I am not sure which type of tests to write to this change and where, as the change affects what CLI is showing, which is mocked in existing tests I've looked at. I can spend more time on tests once we'll be settled with implementation.

Andarist commented on packages/cli/src/commands/add/createChangeset.ts:
```diff
@@ -58,6 +58,14 @@ async function getPackagesToRelease(
+  const pkgJsonsByName = getPkgJsonByName(allPackages);
+
+  // filter out private packages without 'version' field
+  allPackages = allPackages.filter(pkg => !isInternalPackage(pkg.packageJson));
+  changedPackages = changedPackages.filter(
+    pkgName => !isInternalPackage(pkgJsonsByName.get(pkgName)!)
+  );
```
Hm, I guess it makes sense to filter this here - as we might need to keep those packages in the "true" `allPackages` so we can still treat them as internal packages so we can update their dependency ranges etc. The only valid changeset type for them should be `'none'`

bhovhannes replied:
I am not sure I understand you correctly here. Do you mean that we could exclude private packages without `version` field from `allPackages` even before passing `allPackages` to this function, somewhere in `commands/add/index.ts` ?

Andarist replied:
I say that we could exclude them at a different stage - like when getting the list of packages. But that would prevent us from treating those packages as "internal" and thus we couldn't update dependency ranges in them etc. So I think it makes sense to handle this here.

bhovhannes replied:
I see. I was also thinking about that and at the end decided to filter them here, right before showing list of packages to the user. That way filtering affects only the list which cli presents to the user and does not affect other logic.

Andarist commented on packages/cli/src/commands/add/createChangeset.ts:
```diff
@@ -96,6 +104,16 @@ function formatPkgNameAndVersion(pkgName: string, version: string) {
+  );
+}
+
+function isInternalPackage(packageJson: PackageJSON) {
```
We sometimes refer to in-monorepo packages as "internal" so this function name might be a little bit confusing. Maybe this should be changed to smth like `isNonVersionablePackage`?

bhovhannes replied:
Renamed to `isNonVersionablePackage`. It is longer but reflects better what function does. Thanks!

bhovhannes commented:
@Andarist do you have any comments here or this can be merged?

bhovhannes commented:
@Andarist friendly ping. Can this be merged or something else needs to be done first ?

Andarist commented:
@bhovhannes thanks for the ping and sorry for going MIA on this one. I have a couple of things to check in relation to private packages and I would like to land this after I investigate all of them, so my decision can take all of the gathered information into account. For the time being, I think this is OK to go but IIRC somebody has raised concerns that such packages might not be "linkable" in some monorepo setups and I need to check this out.

bhovhannes commented:
> @bhovhannes thanks for the ping and sorry for going MIA on this one. I have a couple of things to check in relation to private packages and I would like to land this after I investigate all of them, so my decision can take all of the gathered information into account. For the time being, I think this is OK to go but IIRC somebody has raised concerns that such packages might not be "linkable" in some monorepo setups and I need to check this out.

I see. Let me know if something will be needed from my side. I plan to migrate one of my monorepos to changesets (pnpm + changesets + turborepo combo) and my monorepo contains private packages without version field, so I'll pay attention to that.

> It seems though that you have rebased this PR in a way that I see now some stuff that shouldn't be here. If you could clean this up - I would highly appreciate it.

rebased again and squashed all my commits into one
